spa_ui/self_service/client/app/states/marketplace/details/details.state.js

```diff
@@ -40,11 +40,45 @@
   }
 
   /** @ngInject */
-  function StateController(dialogs, serviceTemplate) {
+  function StateController($state, CollectionsApi, dialogs, serviceTemplate, Notifications) {
     var vm = this;
 
     vm.title = 'Service Template Details';
     vm.dialogs = dialogs.resources[0].content;
     vm.serviceTemplate = serviceTemplate;
+    vm.submitDialog = submitDialog;
+
+    function submitDialog(vm) {
+      var serviceTemplate = vm.serviceTemplate;
```

Comment:
Similar case here. This function has access to both the `vm.serviceTemplate` and the `serviceTemplate` objects.